rest_device.go

```diff
@@ -376,6 +387,9 @@ func createDeviceRequest(device Device) api.DeviceRequest {
 	req.InterfaceCount = device.InterfaceCount
 	if device.IsSelfManaged != nil {
 		if *device.IsSelfManaged {
+			if *device.Connectivity == Znpd {
```

Comment:
Test is failing due to a panic here.  I think you need a nil check on `device.Connectivity` before dereferencing.

Reply:
Thanks for checking. I updated condition and added test cases